### PR TITLE
Homework week 2 SAEED

### PR DESCRIPTION
**Pull request by SaeedRafay**

### CODE REVIEW

File: Week2/relationships.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      paper_id INT AUTO_INCREMENT PRIMARY KEY,
+      paper_title VARCHAR(100) NOT NULL,
+      conference VARCHAR(100),
+      publish_date DATE
+    );
+  `;
+  const CREATE_TABLE_AUTHORS_PAPERS = `
+    CREATE TABLE IF NOT EXISTS authors_Papers (
+      apid INT AUTO_INCREMENT PRIMARY KEY,
+      author INT,
```
Comment (unmeshvrije) [inline Week2/relationships.js:25]: `NOT NULL` ?

**Review by unmeshvrije — CHANGES_REQUESTED** [review]
good work! Please take a look at some comments

File: Week2/aggregate.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  password: "hyfpassword",
+  database: "userdb",
+});
+
+const execQuery = util.promisify(connection.query.bind(connection));
+
+async function aggregateData() {
+  const ALL_PAPERS_COUNT_AUTHORS = `
+    SELECT p.paper_title, COUNT(DISTINCT ap.author) AS num_of_authors
+    FROM authors_papers AS ap
```
[the remark this answers is not in the document]
Comment (SaeedRafay) [reply Week2/aggregate.js:16]: Thanks 😄

File: Week2/aggregate.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    ;
+  `;
+  const AVRG_HINDX_PER_UNIVERSITY = `
+    SELECT university, AVG(h_index) AS average_h_index
+    FROM authors
+    GROUP BY university
+    ORDER BY average_h_index DESC;
+  `;
+  const COUNT_PAPERS_PER_UNIVERSITY = `
+    SELECT a.university, COUNT(DISTINCT p.paper_title) AS sum_papers
```
[the remark this answers is not in the document]
Comment (SaeedRafay) [reply Week2/aggregate.js]: I have now added `p.paper_id` as well with the title.

File: Week2/aggregate.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  const ALL_PAPERS_COUNT_AUTHORS = `
+    SELECT p.paper_title, COUNT(DISTINCT ap.author) AS num_of_authors
+    FROM authors_papers AS ap
+    JOIN research_papers AS p
+    WHERE p.paper_id = ap.paper
+    GROUP BY ap.paper
+    ORDER BY num_of_authors DESC;
+  `;
+  const ALL_FEMALE_PAPERS_COUNT = `
+    SELECT COUNT(DISTINCT p.paper_title) AS total_female_papers
```
[the remark this answers is not in the document]
Comment (SaeedRafay) [reply Week2/aggregate.js]: I have now added `p.paper_id` as well with the title.

File: Week2/joins.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    a.author_no, a.author_name AS "author", m.author_name AS "mentor"
+    FROM authors AS a
+    JOIN authors AS m
+    ON a.mentor = m.author_no;
+  `;
+  const SELECT_AUTHORS_PAPERS = `
+    SELECT
+    a.author_no, a.author_name, a.university, a.date_of_birth, a.h_index, a.gender, p.paper_title
+    FROM authors AS a
+    LEFT JOIN authors_papers AS ap ON a.author_no = ap.author
```
[the remark this answers is not in the document]
Comment (SaeedRafay) [reply Week2/joins.js]: That was by mistake

File: Week2/relationships.js (hunk 1 above)
Comment (SaeedRafay) [reply Week2/relationships.js]: Added `NOT NULL`. Thanks for notifying.

**Comment by SaeedRafay** [comment]
Thanks, @unmeshvrije for the amazing videos on databases and QnA session and especially for reviewing my homework.

**Pull request closed without merging**